### PR TITLE
upgrade swc to v0.37.0 (Divider, Checkbox , Button , Action button)

### PR DESCRIPTION
**Pull request by bhatiaPankaj**
This PR takes care of upgrading sp-divider and  sp-checkbox,  version 0.37.0 in swc-uxp-wrappers.

[](https://jira.corp.adobe.com/browse/UXP-22033)
[](https://jira.corp.adobe.com/browse/UXP-22041)

### CODE REVIEW

**Comment by inasirhussain** [comment]
@bhatiaPankaj - Please take care of updating packages/divider/README.md and packages/checkbox/README.md to reflect the version upgrade.

File: packages/divider/src/uxp-divider.css [hunk 1]
```diff
@@ -11,3 +11,16 @@ governing permissions and limitations under the License.
 */
 
 /* write uxp style overrides */
+
```
Comment (inasirhussain) [inline packages/divider/src/uxp-divider.css:14]: Please take care of formatting, run 
`yarn format:fix`

Comment (inasirhussain) [reply packages/divider/src/uxp-divider.css:14]: Also remove any unnecessary new line breaks.

File: packages/checkbox/src/uxp-checkbox.css [hunk 2]
```diff
@@ -19,4 +19,12 @@ governing permissions and limitations under the License.
...
     opacity: 0;
+    width: 100%;
+    height: 100%;
 }
+
+#label {
+    margin-top: var(--mod-checkbox-top-to-text,var(--spectrum-checkbox-top-to-text) );
+    margin-left: var(--mod-checkbox-text-to-control,var(--spectrum-checkbox-text-to-control) );
+}
+
```
Comment (inasirhussain) [inline packages/checkbox/src/uxp-checkbox.css:30]: Can you check?, we have to handle and circumvent logical property for id="box" in the sp-checkbox 

`#box {
height: var(
--mod-checkbox-control-size,
var(—spectrum-checkbox-control-size)
);
width: var(
--mod-checkbox-control-size,
var(—spectrum-checkbox-control-size)
);
}
`

Also on host we might have to polyfil the missing logical property pertaining
```suggestion

```

`:host {
max-width: 100%;
min-height: var(—mod-checkbox-height, var(—spectrum-checkbox-height));
}
`

File: packages/button/src/uxp-button.css [hunk 3]
```diff
@@ -11,3 +11,39 @@ governing permissions and limitations under the License.
...
+    );
+    padding-bottom: calc(
+        var(--mod-button-bottom-to-text, var(--spectrum-button-bottom-to-text)) -
+            var(--mod-button-border-width, var(--spectrum-button-border-width))
+    );
+    padding-top: calc(
+        var(--mod-button-top-to-text, var(--spectrum-button-top-to-text)) -
+            var(--mod-button-border-width, var(--spectrum-button-border-width))
+    );
+}
```
Comment (inasirhussain) [inline packages/button/src/uxp-button.css:49]: Please check the case when icon is slotted, we have to handle this case of logical property to be circumvented.

`::slotted([slot="icon"]) {
    margin-inline-start: calc(var(--mod-button-edge-to-visual, var(--spectrum-button-edge-to-visual)) - var(--mod-button-edge-to-text, var(--spectrum-button-edge-to-text)));
}`

**Comment by inasirhussain** [comment]
@bhatiaPankaj - Please revisit, styles for sp-action-button and take care polyfilling these CSSs

::slotted([slot="icon"]) {
    margin-inline-end: calc(var(--mod-actionbutton-edge-to-visual-only, var(--spectrum-actionbutton-edge-to-visual-only) ) - var(--mod-actionbutton-edge-to-text, var(--spectrum-actionbutton-edge-to-text) ));
    margin-inline-start: calc(var(--mod-actionbutton-edge-to-visual, var(--spectrum-actionbutton-edge-to-visual) ) - var(--mod-actionbutton-edge-to-text, var(--spectrum-actionbutton-edge-to-text) ));
}

::slotted([slot="icon"]) {
    max-block-size: 100%;
}

File: projects/swc-starter-webpack/package.json [hunk 4]
```diff
@@ -41,7 +41,7 @@
         "@swc-uxp-wrappers/dialog": "*",
         "@swc-uxp-wrappers/field-group": "*",
         "@swc-uxp-wrappers/button-group": "*",
-        "@spectrum-web-components/theme": "0.14.9",
+        "@spectrum-web-components/theme": "0.35.0",
```
Comment (inasirhussain) [inline projects/swc-starter-webpack/package.json:44]: this need to be 0.37.0

**Comment by inasirhussain** [comment]
@bhatiaPankaj - Try to fix the case where, you have sp-checkbox without label. Currently a margin-left is applied on #label even though there is no label content. 

#label {
    margin-top: var(
        --mod-checkbox-top-to-text,
        var(--spectrum-checkbox-top-to-text)
    );
    margin-left: var(
        --mod-checkbox-text-to-control,
        var(--spectrum-checkbox-text-to-control)
    );
}

**Comment by bhatiaPankaj** [comment]
> @bhatiaPankaj - Try to fix the case where, you have sp-checkbox without label. Currently a margin-left is applied on #label even though there is no label content.
> 
> #label { margin-top: var( --mod-checkbox-top-to-text, var(--spectrum-checkbox-top-to-text) ); margin-left: var( --mod-checkbox-text-to-control, var(--spectrum-checkbox-text-to-control) ); }

The edge case here can not be circumvented via CSS , we can handle this in the card component where this edge case is surfacing.

**Review by inasirhussain — APPROVED** [review]

**Pull request merged**